src/metatensor/models/experimental/gap/model.py

```diff
@@ -0,0 +1,1248 @@
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, Type, Union
+
+import metatensor.torch
+import numpy as np
+import rascaline
+import rascaline.torch
+import scipy
+import skmatter
+import torch
+from metatensor.torch import Labels as TorchLabels
+from metatensor.torch import TensorBlock as TorchTensorBlock
+from metatensor.torch import TensorMap as TorchTensorMap
+from metatensor.torch.atomistic import (
+    MetatensorAtomisticModel,
+    ModelCapabilities,
+    ModelOutput,
+    System,
+)
+from skmatter._selection import _FPS
+
+from metatensor import Labels, TensorBlock, TensorMap
+from metatensor.models.utils.data.dataset import DatasetInfo
+
+from ...utils.export import export
+
+
+class GAP(torch.nn.Module):
+    __supported_devices__ = ["cpu"]
+    __supported_dtypes__ = [torch.float64]
+
+    def __init__(self, model_hypers: Dict, dataset_info: DatasetInfo) -> None:
+        super().__init__()
+
+        if len(dataset_info.targets) > 1:
+            raise NotImplementedError("GAP only supports a single output")
+
+        # Check capabilities
+        for target in dataset_info.targets.values():
+            if target.quantity != "energy":
+                raise ValueError(
+                    "GAP only supports energy-like outputs, "
+                    f"but a {target.quantity} was provided"
+                )
+            if target.per_atom:
+                raise ValueError(
+                    "GAP only supports per-structure outputs, "
+                    "but a per-atom output was provided"
+                )
+        target_name = next(iter(dataset_info.targets.keys()))
+        if dataset_info.targets[target_name].quantity != "energy":
+            raise ValueError("GAP only supports energies as target")
+        if dataset_info.targets[target_name].per_atom:
+            raise ValueError("GAP does not support per-atom energies")
+
+        self.dataset_info = dataset_info
+
+        self.outputs = {
+            key: ModelOutput(
+                quantity=value.quantity,
+                unit=value.unit,
+                per_atom=False,
+            )
+            for key, value in dataset_info.targets.items()
+        }
+
+        self.all_species = dataset_info.atomic_types
```

Comment:
maybe also for consisteny within the code?

```suggestion
        self.atomic_types = dataset_info.atomic_types
```

Reply:
Incidentally this will have to be `sorted()` now

Reply:
`self.atomic_types = sorted(dataset_info.atomic_types)`